packages/server/src/middleware/builder.ts

```diff
@@ -35,12 +35,11 @@ async function checkDevAppLocks(ctx: BBContext) {
   if (!appId || !appId.startsWith(APP_DEV_PREFIX)) {
     return
   }
-  if (!(await doesUserHaveLock(appId, ctx.user))) {
-    ctx.throw(400, "User does not hold app lock.")
-  }
 
-  // they do have lock, update it
-  await updateLock(appId, ctx.user)
+  // If this user already owns the lock, then update it
+  if (await doesUserHaveLock(appId, ctx.user)) {
```

Comment:
Now that we allow multiple users, we don't want to error if the user doesn't have the lock. If they don't have the lock, we just ignore them.

If the user does own the lock, then we update the expiration time.